### ISSUE TITLE
UI Grid 4.0 (Angular 2)

### ISSUE THREAD

**Issue by danicomas** [angular-ui/ui-grid#4294]
We could create a new github project and start to create the angular 2 version.

**Comment by swalters** [comment]
Before we start developing code, we need a design document.  With 3.0, we got ahead of ourselves sometimes by not designing enough and charging into code as proof of concepts.  POC's are necessary, but when it becomes mainline code, it's nothing but trouble.

We need to design how features are added to the grid in Angular 2.  In Angular 1, we used directive stacking.  I don't know enough about Angular 2 to know if that's even possible.  We need to define the extension points for features to make it much more of a concrete API.

We also need to design a good grid render cycle with proper events.

We'll use this doc for now.  We can link in Proof of Concept Plunkers or Gists.
https://docs.google.com/document/d/1YX1RYKrP4WJWdhzJL5t4i4AR_fc2AdAis-gSgiE9X0E/edit?usp=sharing

**Comment by danicomas** [comment]
Nice @swalters

**Comment by danicomas** [comment]
@swalters You can see an example how make a component and access to the DOM https://github.com/angular/angular/tree/8ed22ce6e7ce0e00c12b036d02627424c6b4ff35/modules/angular2_material/src/components/dialog

By now I haven't find a way to add a component without root component or some $compile function. We have to be tuned to new updates

**Comment by forforeach** [comment]
Hi,
Do you have any news on this?
Thanks

**Comment by danicomas** [comment]
Angular 2 beta was released we should work hard to have a new version of ui-grid. @swalters

**Comment by davidrensh** [comment]
Definitely, we are waiting for it.
I have some silly suggestion that you may already knew:
1. I want to start a project to build one thing that similar Power Builder data window. The grid is one of them even it's the #1 important one. But how about anything link to data: dropdown, listbox, menu, navbar
2.UI-grid web-api integration. Why not give an web-api url attribute directly in the grid? We need http call for 90% cases
3. Could it be integrate with Angular Material or kind of support AM. I did a medium size project with AM and UI-Grid. It's perfect. 

Thanks

**Comment by thachp** [comment]
How can I contribute to the development of this milestone?

**Comment by swalters** [comment]
I created a room for angular2 / uiGrid4  https://gitter.im/angular-ui/ui-grid/angular2   so we can discuss.

**Comment by thachp** [comment]
Great. Thank you!  I am happy to contribute.

**Comment by anitapillai** [comment]
Is there a timeline for releasing the Angular 2 version of the ui-grid?

**Comment by JLLeitschuh** [comment]
Nope. This project is almost entirely volunteer. None of us get paid full time to work on this project so most things work on a "its done when it gets finished" basis.
That being said we are always looking for support to move the project forward.
If you or your company is using UI-Grid in your application please consider dedicating some time to contribute and be involved in this project.

**Comment by gomidefabio** [comment]
Hi there! Nice! I'm excited about Angular 2 UI-Grid. Would you guys please post the URL for the Github repo here on this thread or on Gitter once it is initiated? I hope that I will be able to give some contribution. Thank you.

**Comment by vknaveenkumar** [comment]
can anyone say release Date for ui grid ?

**Comment by abhiphirke** [comment]
Is there a tentative release date for UI-GRID 4.0? This is critically important for me to base on it some business decisions!

**Comment by KratiJainImp** [comment]
How can I help with the development? Will be glad to contribute.

**Comment by StarGuardian** [comment]
Where is repository for UI Grid 4.0? Is that project under development?

**Comment by dlgski** [comment]
There is no UI-Grid 4.0 underway just yet.  We are still working through all current PRs and getting developers up-to-speed with what is here today as several of the original owners of this project have moved on to different jobs/projects.

That being said, I would love to see an Angular 2 version getting started in parallel with this one being maintained.  if @KratiJainImp @StarGuardian would like to initiate that, it would be great!

**Comment by webia1** [comment]
Once the day arrives, I would like contribute too,..

**Comment by rahul71** [comment]
Any suggestions of open source grid alternatives for angular 2. We need it ASAP. See a few paid ones out there.

**Comment by Manjunath12-GitHub** [comment]
How can I contribute to this project? I have one year of experience in Angular 1.x and basic understanding of Angular 2

**Comment by sampath-karupakula** [comment]
I would like to contribute... Please give me instructions for this.....

**Comment by kpats** [comment]
Any update on this? The final version of Angular 2 is released.. Really looking forward work getting started on UI Grid 4.0

**Comment by dsaf** [comment]
Angular 2 is not really released then. GUI framework without a data-grid is a joke.

**Comment by zalayeto** [comment]
Ready to take part as well. Who is supposed to kick off this?

**Comment by anvesh48** [comment]
I am interested to contribute.Please let me know.

**Comment by shafqatdotnet** [comment]
When its going to start?, now angular 2.01 is also out.

**Comment by slamflipstrom** [comment]
Any movement on this? Our team would like to continue with UI Grid, but we're looking into alternatives since there haven't been any recent updates.

**Comment by csvan** [comment]
+1, also willing to contribute

**Comment by ezzkht** [comment]
Unfortunately, I don't see any progress on this project! I guess it's time to switch to alternatives.

**Comment by csvan** [comment]
@ezzkht or you could, well, help out. Open source and all that.

**Comment by ezzkht** [comment]
I'm willing to help, for sure. Just I can't see any progress on the plan so far from the initiative people!

**Comment by JT-Bruch** [comment]
Has anyone had any luck wrapping Ui-Grid inside of a NG2 component?

**Comment by ezzkht** [comment]
You need to wrap the ui-grid with an angular 1 component first, as the fact that ui-grid violates most rules of the ng-upgrade (the compile command for instance). After wrapping, just upgrade the angular 1 wrapper using ng-upgrade. Finally, you'll be able to use that wrapper from within angular 2 component.

**Comment by JT-Bruch** [comment]
I was afraid of that... I think finding another open source grid that is supported miiiight be easier. 

I found this one and it has all the features we're looking for and its very mimisticially styled so it looks quite attractive as of this point.

http://www.primefaces.org/primeng/#/datatable

**Comment by ezzkht** [comment]
You're absolutely right! I took the short path and replaced the ui-grid with primefaces datatable. I've searched all over the net, and found that the best suitable replacement, which is full of features and options.
Good luck with your migration :)

**Comment by alexkostenko** [comment]
Cmon guys how long we should wait?

**Comment by Srini25vasu** [comment]
I'm willing to help UI Grid 4.0 with angular 2.  can you please provide some instructions?

**Comment by ZiadJ** [comment]
@mportuga said this on [https://github.com/angular-ui/ui-grid/issues/5817](https://github.com/angular-ui/ui-grid/issues/5817):
> We have not attempted to make UI-Grid work with angular 2 or ES6. In the future, please direct support queries to gitter rather than logging an issue.

From that comment I guess I can safely conclude that we're not going to get there anytime soon and it's about time to look for alternatives, like the ones mentioned above, if we're to make the switch.

**Comment by webia1** [comment]
I miss ui-grid; there is nothing comparable existing for Angular2 :cry:

**Comment by anvesh48** [comment]
@webia1 Have you checked ag-grid?

**Comment by csvan** [comment]
ag-grid is proprietary (I am just assuming people looking for an alternative is expecting an open source one).

**Comment by nikiben** [comment]
There's another alternative option that's comparable. Like @JT brunch mentioned check out primeng. It has nice features as well

http://www.primefaces.org/primeng/#/datatable

**Comment by SatyabratKumarSingh** [comment]
Sorry anyone news on UI-grid for Angular 2? I should be able to contribute as well if anyone plans to start it.... 
Isn't there any easy way to customize it for angular 2 apps ?

**Comment by Zyprexia** [comment]
Actually, Telerik has recently released a RC0 NG2 package for their Kendo UI framework. It's feature stack is comparable to the original UI-grid. I am inclined to mention that Kendo UI is a commercial product. I know that there is also a 'community' edition of Kendo UI (which is free and open source), but am unaware of those include NG2 bindings.

For more information, visit: http://www.telerik.com/kendo-angular-ui/components/grid/configuration/

**Comment by JosephSKh** [comment]
guys, have any of you tried this?
[https://github.com/vladotesanovic/ui-grid-angular2](https://github.com/vladotesanovic/ui-grid-angular2)

**Comment by kpats** [comment]
@JosephSKh Seems working correctly. has anyone else given it a try? Any pros and cons for the approach taken?

**Comment by digioz** [comment]
So is there a time frame for an Angular 2 UI Grid as of right now? I love the current one and it would be great to take advantage of some of the Angular 2 speed and rendering improvements.

**Comment by JosephSKh** [comment]
@kpats not yet actually, I tried it it's functioning the main functions, but didn't dive deep
,,, yes we are in a great need of a time-frame

**Comment by biopete** [comment]
How are people liking primefaces  primeng datatables? Any comparisons or anything lacking you have found or anything better you have found? I love ui-grid because it is easy to work with, edit works good,  and has done everything we need so far.

**Comment by prksean** [comment]
This may be an ignorant question but can we use ui-grid with Angular2 while we wait for ui-grid4 to release? I miss using ui-grid with AngularJS so much..

**Comment by danicomas** [comment]
@prksean exists a bridge using @angular/upgrade, however ui-grid is not agnostic using $scope in $on events. It should be fixed. Meanwhile you could check this plnkr: http://plnkr.co/edit/HE1mK3?p=info

Issue: https://github.com/angular-ui/ui-grid/issues/6091

**Comment by lobo-tuerto** [comment]
I don't know about you guys, but while this comes out I've found https://github.com/swimlane/ngx-datatable a pleasure to work with.

**Comment by JosephSKh** [comment]
@lobo-tuerto I've tried it and it's full of issues, I had some difficulties dealing with it,
I've implemented this on my part
https://github.com/JosephSKh/j-ng2-table
would love for you guys to try it and tell me enhancements, improvements and issues.

**Comment by mjedwabn** [comment]
Do we have any update on this? Roadmap, plans...?

**Comment by lobo-tuerto** [comment]
Angular Material team has the first version of their own `data-table` implementation out: https://material.angular.io/components/table/overview

**Comment by JosephSKh** [comment]
@lobo-tuerto it still needs some more features to be useful but I can see great potential
Thanks for mentioning

**Comment by UtkarshBais** [comment]
Hi Guys, I've been working with the Angular UI Grid with Angular 1.6 for sometime, I would like to contribute on the theme side.

**Comment by mjedwabn** [comment]
IMHO all Material components tend to be lightweight by design. I wouldn't expect that data-**table** evolve to a data **grid**. So I will stick to my origin question - what plans do authors/main contributors have for ui-grid in context of Angular 2+?

**Comment by sancelot** [comment]
There is really a need for migration to angular 2. Since there is no activity regarding this question since aug. 2015 , I think a fork is needed

**Comment by sumanjara** [comment]
is UI GRID released for angular version 4.0?If so please share one example for Ui-Grid for angular version 4.0 or angular version 2.0

**Comment by akashdeepg12** [comment]
angular 4 version  of ui-grid is avail  ? If so please share details.
primeNg angular 4 are not supported many feature as avail in UI-grid(angular1).
we really need ui-grid

**Comment by lobo-tuerto** [comment]
For what is worth, the Angular Material has released some components for building a data table, alongside some basic ones to build your own implementation if you want: https://material.angular.io/components/table/overview

**Comment by clystian** [comment]
this repo tries migrate ui-grid to NG2 https://github.com/vladotesanovic/ui-grid-angular2 
@JosephSKh nice start with https://github.com/JosephSKh/j-ng2-table

**Comment by anoop250688** [comment]
Hi everyone, Is it possible to use ui-grid in Angular-4, If it possible can anyone please share me the code.

**Comment by tramel-woodard** [comment]
Here is a promising data table directive (component) for Angular 4:

[Angular 4 Data Table - Bootstrap 4](https://github.com/afermon/angular-4-data-table-bootstrap-4)

You just need to run **`npm install angular-4-data-table-bootstrap-4 --save`** into your Angular 4 project.

There are three active project demos on this page:

[https://afermon.github.io/angular-4-data-table-bootstrap-4-demo/](https://afermon.github.io/angular-4-data-table-bootstrap-4-demo/)

and the three respective demo repos on this page:

[https://github.com/afermon/angular-4-data-table-bootstrap-4-demo/tree/master/app](https://github.com/afermon/angular-4-data-table-bootstrap-4-demo/tree/master/app)

Looks promising... currently in the process of using them with my own project and will update on the progress or any issues that arise.

**Comment by burfom88** [comment]
Alternatively, PrimeNG seems to have a very decent data table module with a lot of addon features. 

https://www.primefaces.org/primeng/#/datatable

**Comment by nsksaisaravana** [comment]
Hi everyone, please make some progress on this awesome component in Angular 2 +, I have evaluated other grid components but nothing work like this. I am missing this in Angular 2 +.

**Comment by mportuga** [comment]
We currently have a plan to stabilize the grid and then start developing a version for angular 5+.
See it here: https://github.com/angular-ui/ui-grid/blob/master/planning_5.0.md

**Comment by nsksaisaravana** [comment]
Very good news!! We are missing this in Angular. Please start ASAP.

**Comment by HaakonH** [comment]
For the love of all that is Angular, please please please... and Thank You! :)

**Comment by portiss** [comment]
I'de like to have this release soon as well! tnxalot:)

**Comment by nsksaisaravana** [comment]
@mportuga We are waiting for this for  a long time, Please announce some date.

**Comment by csvan** [comment]
@nsksaisaravana please contribute a few PRs rather than expecting someone else to do all the work.

**Comment by sebastian-zarzycki-es** [comment]
What a naive and juvenile approach. Add me to the ranks of those expecting someone else to do all the work.

**Comment by ghiscoding** [comment]
Hello everyone, I used to have UI-Grid in few of our projects as well, but the lack of Angular 2+ support made me start my own library. So you guys  might be interested in looking at [Angular-Slickgrid](https://github.com/ghiscoding/Angular-Slickgrid) which is a wrapper of the amazingly fast jQuery lib [SlickGrid](https://github.com/mleibman/SlickGrid). It's been out for almost a year, so it's quite stable now.

UI Grid is still amazing nonetheless

**Comment by jameelgrand** [comment]
any updates on angular2 support ?? waiting from 2 years I need to upgrade my project to angular 2, heavily depended on uiGrid

**Comment by digioz** [comment]
@jameelgrand - Here we are 4 years later and we are still asking this same question. I am in the same boat as you. I used this component in a critical project and now I am stuck in Angular 1 because of it. 

At this point if the development team ever does want to produce a more modern version of this Grid they should target Angular 8. I am not holding my breath on that though, since as mentioned before this project is purely volunteer driven and therefore cannot be held accountable to any deadlines or time frames. 

Personally I am planning to move my project to Kendo UI despite the additional effort required to do so. It does have its disadvantages, but their development schedule is a lot more predictable.

**Comment by itis5** [comment]
@digioz @jameelgrand Likewise. I've been using this since first project but next project requires me to move on to Angular 7+. That being said, I may check out the Kendo UI as well. Anyway, I am hoping to see this wonderful UI-Grid in Angular 7+.

**Comment by ghiscoding** [comment]
@itis5, @digioz, @jameelgrand  and the others...
Have you looked at my lib [Angular-Slickgrid](https://github.com/ghiscoding/Angular-Slickgrid)? It's seriously the closest you could get to UI-Grid implementation. It's usage is very similar to UI-Grid as well and I did use UI-Grid in the past, so I implemented some of their ways of doing things. The reason I created the lib (which is a wrapper of [SlickGrid](https://github.com/mleibman/SlickGrid) a jQuery datagrid) was because I couldn't find anything similar to UI-Grid and all our projects requires an enterprise ready data grid. There's a ton of Wikis and [demos](https://ghiscoding.github.io/Angular-Slickgrid/#/basic), there's a bunch of unit tests (nearly 75% coverage)... and it's fully Open Source. 

As for UI-Grid, I don't want to be pessimistic, but in order to support Angular 2+, it would most probably require a complete rewrite of UI-Grid and I would be rather surprised if that ever happens. 

As for Kendo UI it isn't free, the other option would be AG-Grid (they have a free version but the best features are only in the paid version). Anyhow good luck and happy coding.

**Comment by Simple2012hcz** [comment]
@ghiscoding ,the demo is perfect. We will ugrade angularjs to angular8, but no offical docs about ui-grid in angular8, I still hope the planning is coming.

**Comment by richardn739** [comment]
Since Slickgrid is no longer being developed, is there a jQuery or javascript based grid that is being supported and updated? If so could we put an Angular wrapper around that such as @ghiscoding did around Slickgrid? An Angular 2 ui-grid would be great but I'm not holding my breath.

**Comment by ghiscoding** [comment]
> Since Slickgrid is no longer being developed
@richardn739  I don't know where you got that, unless you're talking about the original repo? The 6pac/SlickGrid fork is more than alive. We worked on fixing lots of bugs and added lots of new features/plugins to the lib, I have personally worked on adding Grid Menu, Row Detail, Cell Menu, Context Menu, Tree Data and Frozen Grids... no longer being developed??? Think again... it's more alive now that it was even before. The 6pac repo is still the core library that I use to build my libs, there's no reason to rewrite something that already exist and is proven. I now have 3 wrappers (frameworks) on top of the 6pac/SlickGrid fork (all done in TypeScript). SlickGrid existed long before any other grids and it's still the fastest, UI-Grid probably copied a few pieces of the structure of SlickGrid, which is a good thing since Angular-Slickgrid is the closest you'll get to UI-Grid (I also used similar patterns of UI-Grid, so there's that too). I don't have anything to prove anymore, Angular-Slickgrid has over 250 users, the only thing I wish would be to contributors, it's a lot to handle for 1 person (me that is). 

> An Angular 2 ui-grid would be great but I'm not holding my breath.

If it didn't happen until now, it will most probably never happen unless the entire UI-Grid is rewritten (thanks to the fact that Angular is totally different than AngularJS, I still think it should have been renamed with a totally different name, they only kept the name for marketing purposes). The official AngularJS End of Life is June of 2021, the countdown has already started.

You also have a paid option, Ag-Grid is another alternative but it's not free... actually there is the community one but it's missing so many features that it's not worth using unless you want a simple grid.
